### PR TITLE
8281314: Rename Stack{Red,Yellow,Reserved,Shadow}Pages multipliers

### PR DESCRIPTION
**Pull request by shipilev**
This was a question during [JDK-8072070](https://bugs.openjdk.java.net/browse/JDK-8072070) review. The code in `StackOverflow::initialize_stack_zone_sizes()` multiplies `Stack{Red,Yellow,Reserved,Shadow}Pages` by "alignment", while that is actually just a "unit".

```
  product_pd(intx, StackYellowPages, \
          "Number of yellow zone (recoverable overflows) pages of size " \
          "4KB. If pages are bigger yellow zone is aligned up.") \
          range(MIN_STACK_YELLOW_PAGES, (DEFAULT_STACK_YELLOW_PAGES+5)) \
```

We can rename it for clarity.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8281314](https://bugs.openjdk.java.net/browse/JDK-8281314): Rename Stack{Red,Yellow,Reserved,Shadow}Pages multipliers

### Reviewers
 * [Thomas Stuefe](https://openjdk.java.net/census#stuefe) (@tstuefe - **Reviewer**) ⚠️ Review applies to 608ca21ac60fdc6d57542f146e122132af62d64e
 * [Coleen Phillimore](https://openjdk.java.net/census#coleenp) (@coleenp - **Reviewer**)
 * [Xin Liu](https://openjdk.java.net/census#xliu) (@navyxliu - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7362/head:pull/7362` \
`$ git checkout pull/7362`

Update a local copy of the PR: \
`$ git checkout pull/7362` \
`$ git pull https://git.openjdk.java.net/jdk pull/7362/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7362`

View PR using the GUI difftool: \
`$ git pr show -t 7362`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7362.diff">https://git.openjdk.java.net/jdk/pull/7362.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/runtime/stackOverflow.cpp [hunk 1]
```diff
@@ -42,16 +42,16 @@ void StackOverflow::initialize_stack_zone_sizes() {
   // We need to adapt the configured number of stack protection pages given
   // in 4K pages to the actual os page size. We must do this before setting
   // up minimal stack sizes etc. in os::init_2().
-  size_t alignment = 4*K;
+  size_t unit = 4*K;
```
[the remark this answers is not in the document]
Comment (shipilev) [reply src/hotspot/share/runtime/stackOverflow.cpp:46]: ...like in the new commit.

File: src/hotspot/share/runtime/stackOverflow.cpp [hunk 2]
```diff
@@ -39,9 +39,10 @@ void StackOverflow::initialize_stack_zone_sizes() {
   // Stack zone sizes must be page aligned.
   size_t page_size = os::vm_page_size();
 
-  // We need to adapt the configured number of stack protection pages given
-  // in 4K pages to the actual os page size. We must do this before setting
-  // up minimal stack sizes etc. in os::init_2().
+  // We need to adapt the configured number of stack protection pages to the
+  // actual OS page size. We must do this before setting up minimal stack
+  // sizes etc. in os::init_2(). The option values are given in the 4K units,
```
Comment (tstuefe) [inline src/hotspot/share/runtime/stackOverflow.cpp:44]: s/in the 4K units/in 4K units

**Review by tstuefe — APPROVED** [review]
LGTM

File: src/hotspot/share/runtime/stackOverflow.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (GoeLin) [reply src/hotspot/share/runtime/stackOverflow.cpp:46]: Good point to rename this!

[the remark this answers is not in the document]
Comment (coleenp) [reply src/hotspot/share/runtime/stackOverflow.cpp:46]: Oh, ok, reading the option description helps.

**Review by coleenp — APPROVED** [review]
Approved and this is trivial.

**Review by navyxliu — APPROVED** [review]
LGTM. I am not a reviewer, we need other reviewers to approve it. 

now I see. the number 4k is deliberately chosen as 'unit', not page size. The original code was correct, but it's better to understand now.

**Comment by shipilev** [comment]
Thanks! The test failure in GHA is unrelated.

/integrate

**Pull request closed without merging**